### ISSUE TITLE
OAuthPrompt login is stuck on Microsoft Teams Android

### ISSUE THREAD

**Issue by claudiucristeagab** [microsoft/botframework-sdk#6557]
## Version
Microsoft.Bot.Builder: 4.15.2 (dotnet)
Microsoft.Bot.Builder.Dialogs: 4.15.2 (dotnet)

## Describe the bug
Channel: Microsoft Teams on Android
OS: Android 13
Phone: Samsung S22 Ultra

Logging in using the OAuthPrompt in the Microsoft Teams app from an Android phone does not work properly. Clicking on the login button and after successfully logging in, the in-app browser automatically closes and the prompt remains stuck. Writing anything triggers the reprompt message.
This issue could not be reproduced on Microsoft Teams on Windows, Microsoft Teams on iOS or Facebook Messenger.

## To Reproduce
1. Enter into a dialog with an OAuthPrompt.
2. Click on the Login button inside the card.
3. Enter your login info inside the inside browser and login.
4. Prompt remains stuck.

## Expected behavior
The OAuthPrompt will automatically continue and finish after I successfully log in.

## Screenshots
![Screenshot_20221213_161552_Teams](https://user-images.githubusercontent.com/32666529/207357654-38e1cd0c-1a7a-42dc-bd33-f2fd4efbe8b2.jpg)

## Tracking Status

**Dotnet SDK** [TODO]()
- [ ] PR
- [ ] Merged

**Javascript SDK** [TODO]()
- [ ] PR
- [ ] Merged

**Python SDK** [TODO]()
- [ ] PR
- [ ] Merged

**Java SDK** [TODO]()
- [ ] PR
- [ ] Merged

**Samples** [TODO]()
- [ ] PR
- [ ] Merged

**Docs** [TODO]()
- [ ] PR
- [ ] Merged

**Tools** [TODO]()
- [ ] PR
- [ ] Merged

**Comment by dmvtech** [comment]
I am investigating to see if I can reproduce this and troubleshoot.

**Comment by dmvtech** [comment]
Sorry for the delay. Still investigating. Hope to have an update shortly.

**Comment by dmvtech** [comment]
I am able to reproduce this issue. I have a feeling that the authentication isn't actually successful (or the token isn't successfully applied/kept in the bot).

![Screenshot_20230105-094831_Teams](https://user-images.githubusercontent.com/43397658/210867502-0ba16eb9-bc57-4727-98a7-d5665bc0a498.jpg)

I also did try some odd stuff like asking for the oauth prompt in Android app, but then clicking login button from within the browser Teams page. And received the error `InputDialog is missing Prompt`.

![image](https://user-images.githubusercontent.com/43397658/210867865-9f313e2d-57a4-4c82-926f-5ea75b9e367c.png)
 
 
 But then re-visiting the prompt showed that I was logged in.
 
![image](https://user-images.githubusercontent.com/43397658/210868189-a5f66d8a-5930-4a8b-b700-16b3705fa7c9.png)

At some point, between juggling between prompts in Android and browser, received the following: `An item with the same key has already been added. Key: BotFrameworkAdapter.InvokeResponse`

@juanar Any thoughts on this one?

**Comment by sw-joelmut** [comment]
We were able to reproduce the issue using Android 9 and both .NET and JS bots.
We tested the behavior in three platforms (Browser, Windows and Android), and we noticed that in Android, MS Teams
is not returning the magic code in the activity's value property after clicking in the SignIn button of the [OAuthPrompt](https://github.com/microsoft/botbuilder-dotnet/blob/fe6f8489d0a8e9c64c601323c8117e9f267883e0/libraries/Microsoft.Bot.Builder.Dialogs/Prompts/OAuthPrompt.cs#L59) card.
In conclusion, we believe that this is an issue in the MS Teams Android app.
![image](https://user-images.githubusercontent.com/62260472/211882296-89e11315-d9a1-4b3e-af2e-549cfd4e607f.png)
![image](https://user-images.githubusercontent.com/62260472/211882309-23c16b04-7371-4cac-9761-55c0da8ff89b.png)

**Comment by dmvtech** [comment]
@claudiucristeagab As we have identified this as a bug with the Teams Android app and not the Bot Framework SDK, we are closing this. Please pursue opening a bug/support ticket within Teams/Android client app. Feel free to reference this issue.
